.github/workflows/ci.yml

```
@@ -3,15 +3,24 @@ on:
     branches:
       - master
   pull_request:
+
 name: CI
 jobs:
   test:
     name: Test
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: ["1.17", "1.18", "1.19", "1.20", "1.21", "1.22", "stable"]
     steps:
-      - name: Checkout code
+      - name: Checkout Repository
         uses: actions/checkout@v4
-      - name: Init Hermit
-        run: ./bin/hermit env -r >> $GITHUB_ENV
+
+      - name: Setup Golang Environment
+        uses: actions/setup-go@v5
```

Minor nit, It's recommended to pin this to a git hash for supply chain security.

Renovate can also keep this up to date.
```suggestion
        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
```

---

yeah but this repo is not really maintained...I'm not sure who would merge all the PRs...

---

I mean I’d be down to help maintain it

---

@alecthomas ?

---

I appreciate the effort but I personally don't think it's useful to test on historical versions of Go at all, for multiple reasons:

1. Only the most recent 2 versions are supported by the Go team, and I generally adhere to this policy too.
2. The Go backwards compatibility promise is incredibly reliable.

The other changes seem reasonable.

---

My usual recommendation here is to update the `go` directive in the `go.mod` file to support the latest 2 releases. This way the matrix test matches. This is the policy we use for most of the Prometheus ecosystem.

---

I changed it to only test `oldstable` and `stable`.